File: ReadingNotes/CythonDocumentation/prime/prime.cpp
// 
// @Filename     :prime.cpp
// @Author       :Arthur Zhan
// @Init Time    :2020/06/16
// 


#include <iostream>
#include <vector>
#include <chrono>
#include "prime.h"

PRIME::PRIME(unsigned int n)
{
    num = n;
}

void PRIME::calculate()
{
    unsigned int i = 2;
    unsigned int limit;
    bool test = true;
    while (list.size() < num)
    {
        if (2 > i / 2 + 1)
        {
            limit = i;
        }
        else
        {
            limit = i / 2 + 1;
        }
        for (unsigned int d = 2; d < limit; ++d)
        {
            if (i % d == 0)
            {
                test = false;
                break;
            }
            else
            {
                test = true;
            }
        }
        if (test)
        {
            list.push_back(i);
        }
        i++;
    }
}

void PRIME::show()
{
    for (long unsigned int i = 0; i < list.size(); ++i)
    {
        std::cout << list[i] << " ";
    }
    std::cout << std::endl;
}

File: ReadingNotes/CythonDocumentation/prime/prime.py
#!/usr/bin/env python
# -*- encoding: utf-8 -*-
"""
@Filename     :prime_origin.py
@Author       :Arthur Zhan
@Init Time    :2020/06/16
"""


def primes_1(num):
    i = 2
    lst = []
    while len(lst) < num:
        if 2 > i // 2 + 1:
            limit = i
        else:
            limit = i // 2 + 1
        for d in range(2, limit):
            if i % d == 0:
                break
        else:
            lst.append(i)
        i += 1
    return lst


def primes_2(num):  # According to documentation's example
    i = 2
    lst = []
    while len(lst) < num:
        for d in lst:
            if i % d == 0:
                break
        else:
            lst.append(i)
        i += 1
    return lst
